api/controllers/records.js

```diff
@@ -1,13 +1,22 @@
-var recordUtils = require('./record-utils');
+const db = require('../db-pouch'),
+      recordUtils = require('./record-utils');
+
+const generate = (req, contentType) => {
+  if (contentType === 'urlencoded') {
+    const options = { locale: req.query && req.query.locale };
+    return recordUtils.createByForm(req.body, options);
+  }
+  if (contentType === 'json') {
+    return recordUtils.createRecordByJSON(req.body);
+  }
+};
 
 module.exports = {
-  create: function(data, contentType, callback) {
-    if (contentType === 'urlencoded') {
-      return recordUtils.createByForm(data, callback);
-    }
-    if (contentType === 'json') {
-      return recordUtils.createRecordByJSON(data, callback);
+  create: (req, contentType) => {
+    const doc = generate(req, contentType);
+    if (!doc) {
+      return Promise.reject(new Error('Content type not supported.'));
```

Comment:
I think this error would make more sense being thrown inside `generate()`

Reply:
Done.